com.unity.sg2/Editor/GraphUI/DataModel/GraphDataNodeModel.cs

```diff
@@ -172,10 +172,24 @@ public int dismissedUpgradeVersion
 
         internal int currentVersion => registryKey.Version;
 
-        internal int latestAvailableVersion => graphHandler.registry.BrowseRegistryKeys()
-            .Where(otherKey => otherKey.Name == registryKey.Name)
-            .Select(otherKey => otherKey.Version)
-            .Max();
+        internal int latestAvailableVersion
+        {
+            get
+            {
+                try
+                {
+                    return graphHandler.registry.BrowseRegistryKeys()
```

Comment:
@dhsavell You'll note that I added this try/catch here, I needed to because this would randomly pop up when trying to delete node and edge selections but it doesn't seem to break anything else

I think its got something to do with a node being in an invalid state around deletion time? But it's not consistent repro, not sure what triggers it exactly, but I wanted to add this for more visibility

Reply:
Good find, I often forget about exceptions when using the linq extensions. It might be because the name checks if the node is out-of-date and appends the version number if it is.

"_0" seems to be the string representation of an empty registry key, so my guess is that something's querying the version of a node that doesn't have actual backing data.

Anyway, I'd just see if making the fallback value -1 (failing that, 0) works. It looks like most node versions start at 0 so the fallback should be <= 0 to prevent upgrade prompts from appearing, on the off chance we get in that state.